### PR TITLE
MDS-495 | Explicit audit event changes

### PR DESCRIPTION
**Pull request by matthewharris1**

### CODE REVIEW

File: test/unit/uk/gov/hmrc/individualsbenefitsandcreditsapi/audit/AuditHelperSpec.scala [hunk 1]
```diff
@@ -50,84 +53,71 @@ class AuditHelperSpec extends UnitSpec with MockitoSugar {
...
-      capturedEvent.asInstanceOf[ExtendedDataEvent].detail shouldBe result
+      val captor = ArgumentCaptor.forClass(classOf[ScopesAuditEventModel])
+
+      auditHelper.auditAuthScopes(matchId, scopes, request)
+
+      verify(auditConnector, times(1)).sendExplicitAudit(eqTo("AuthScopesAuditEvent"),
+        captor.capture())(any(), any(), any())
+
+      val capturedEvent = captor.getValue.asInstanceOf[ScopesAuditEventModel]
+      capturedEvent.asInstanceOf[ScopesAuditEventModel].apiVersion shouldEqual "1.0"
```
Comment (grab-a-byte) [inline test/unit/uk/gov/hmrc/individualsbenefitsandcreditsapi/audit/AuditHelperSpec.scala]: capturedEvent is already instance of ScopesAuditEventModel (from line above)

File: test/unit/uk/gov/hmrc/individualsbenefitsandcreditsapi/audit/AuditHelperSpec.scala [hunk 2]
```diff
@@ -137,70 +127,37 @@ class AuditHelperSpec extends UnitSpec with MockitoSugar {
...
-      capturedEvent.asInstanceOf[ExtendedDataEvent].detail shouldBe result
+      val captor = ArgumentCaptor.forClass(classOf[IfApiResponseEventModel])
+
+      auditHelper.auditIfApiResponse(correlationId, matchId, request, ifUrl, ifResponse)
+
+      verify(auditConnector, times(1)).sendExplicitAudit(eqTo("IfApiResponseEvent"),
+        captor.capture())(any(), any(), any())
+
+      val capturedEvent = captor.getValue.asInstanceOf[IfApiResponseEventModel]
+      capturedEvent.asInstanceOf[IfApiResponseEventModel].matchId shouldEqual matchId
```
Comment (grab-a-byte) [inline test/unit/uk/gov/hmrc/individualsbenefitsandcreditsapi/audit/AuditHelperSpec.scala]: capturedEvent is already an IfApiResponseEventModel from line above

**Review by ajchilts — COMMENTED** [review]
👍  - when review comments addressed @matthewharris1

Where you have already specified the captor type you just need to do

`captor.getValue `

In place of 

`captor.getValue.asInstanceOf[IfApiResponseEventModel]`

File: test/unit/uk/gov/hmrc/individualsbenefitsandcreditsapi/audit/AuditHelperSpec.scala (hunk 1 above)
Comment (matthewharris1) [reply test/unit/uk/gov/hmrc/individualsbenefitsandcreditsapi/audit/AuditHelperSpec.scala]: Good spot

File: test/unit/uk/gov/hmrc/individualsbenefitsandcreditsapi/audit/AuditHelperSpec.scala (hunk 2 above)
Comment (matthewharris1) [reply test/unit/uk/gov/hmrc/individualsbenefitsandcreditsapi/audit/AuditHelperSpec.scala]: Good spot

**Review by ajchilts — APPROVED** [review]

File: app/uk/gov/hmrc/individualsbenefitsandcreditsapi/controllers/ChildTaxCreditController.scala [hunk 3]
```diff
@@ -57,8 +57,8 @@ abstract class ChildTaxCreditController @Inject()(
               val selfLink = HalLink("self", urlWithInterval(s"/individuals/benefits-and-credits/child-tax-credits?matchId=$matchId", interval.getStart))
               val response = Json.obj("applications" -> Json.toJson(applications))
 
-              auditHelper.auditApiResponse(correlationId.toString, matchId.toString, Some(authScopes.mkString(",")),
-                request, selfLink.toString, Json.toJson(response))
+              auditHelper.childTaxCreditAuditApiResponse(correlationId.toString, matchId.toString, authScopes.mkString(","),
+                request, response.toString(), applications)
```
Comment (ajchilts) [inline app/uk/gov/hmrc/individualsbenefitsandcreditsapi/controllers/ChildTaxCreditController.scala]: Please only pass the selfLink object here or the self links will contain the entire response object.

File: app/uk/gov/hmrc/individualsbenefitsandcreditsapi/controllers/WorkingTaxCreditController.scala [hunk 4]
```diff
@@ -55,8 +55,8 @@ abstract class WorkingTaxCreditController @Inject()(
               val selfLink = HalLink("self", urlWithInterval(s"/individuals/benefits-and-credits/working-tax-credits?matchId=$matchId", interval.getStart))
               val response = Json.obj("applications" -> Json.toJson(applications))
 
-              auditHelper.auditApiResponse(correlationId.toString, matchId.toString, Some(authScopes.mkString(",")),
-                request, selfLink.toString, Json.toJson(response))
+              auditHelper.workingTaxCreditAuditApiResponse(correlationId.toString, matchId.toString, authScopes.mkString(","),
+                request, response.toString, applications)
```
Comment (ajchilts) [inline app/uk/gov/hmrc/individualsbenefitsandcreditsapi/controllers/WorkingTaxCreditController.scala]: Please only pass the selfLink object here or the self links will contain the entire response object.

**Review by ajchilts — CHANGES_REQUESTED** [review]

**Review by ajchilts — APPROVED** [review]

**Pull request merged**